s3psetup/clean_open_ldap_by_s3.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+die_with_error () {
+        echo $1
+        exit -1
+}
+
+# Attempt ldap clean up since ansible openldap setup is not idempotent
+systemctl stop slapd 2>/dev/null || die_with_error "slapd could not be stopped!"
+yum remove -y openldap-servers openldap-clients || "openldap-servers openldap-clients removal failed"
```

Comment:
Can you remove this line. Not required to remove the package during cleanup.

Reply:
This script is not part of the production code. Last friday you and Amit suggested to keep the steps that needs to be run to cleanup ldap accounts in a separate script so that anyone needs to run cleanup and rerun setup can manually run the steps in between until s3_setup becomes completely idempotent.

Reply:
yes agreed, but removing ldap packages is not required for cleanup purpose. This was tested I think. Just confirm once if you can re-configure ldap without removing the packages. If so we can remove that line.

Reply:
After those slapd config files are removed, slapd fails to start.

[root@ssc-vm-1735 ~]# systemctl stop slapd
[root@ssc-vm-1735 ~]# systemctl start slapd
Job for slapd.service failed because a configured resource limit was exceeded. See "systemctl status slapd.service" and "journalctl -xe" for details.
[root@ssc-vm-1735 ~]# systemctl status slapd.service
● slapd.service - OpenLDAP Server Daemon
   Loaded: loaded (/usr/lib/systemd/system/slapd.service; enabled; vendor preset: disabled)
   Active: failed (Result: resources) since Mon 2021-01-11 04:24:22 MST; 41s ago
     Docs: man:slapd
           man:slapd-config
           man:slapd-hdb
           man:slapd-mdb
           file:///usr/share/doc/openldap-servers/guide.html
  Process: 34221 ExecStart=/usr/sbin/slapd -u ldap -h ${SLAPD_URLS} $SLAPD_OPTIONS (code=exited, status=0/SUCCESS)
  Process: 34192 ExecStartPre=/usr/libexec/openldap/check-config.sh (code=exited, status=0/SUCCESS)
 Main PID: 34223 (code=exited, status=0/SUCCESS)

Jan 11 04:24:22 ssc-vm-1735.colo.seagate.com systemd[1]: Stopped OpenLDAP Server Daemon.
Jan 11 04:24:22 ssc-vm-1735.colo.seagate.com systemd[1]: Failed to load environment files: No such file or directory
Jan 11 04:24:22 ssc-vm-1735.colo.seagate.com systemd[1]: slapd.service failed to run 'start-pre' task: No such file or directory
Jan 11 04:24:22 ssc-vm-1735.colo.seagate.com systemd[1]: Failed to start OpenLDAP Server Daemon.
Jan 11 04:24:22 ssc-vm-1735.colo.seagate.com systemd[1]: Unit slapd.service entered failed state.
Jan 11 04:24:22 ssc-vm-1735.colo.seagate.com systemd[1]: slapd.service failed.
Jan 11 04:24:58 ssc-vm-1735.colo.seagate.com systemd[1]: Failed to load environment files: No such file or directory
Jan 11 04:24:58 ssc-vm-1735.colo.seagate.com systemd[1]: slapd.service failed to run 'start-pre' task: No such file or directory
Jan 11 04:24:58 ssc-vm-1735.colo.seagate.com systemd[1]: Failed to start OpenLDAP Server Daemon.
Jan 11 04:24:58 ssc-vm-1735.colo.seagate.com systemd[1]: slapd.service failed.


Those open ldap packages needs to be removed and reinstalled to get around this problem.

Reply:
openldap-servers, openldap-clients rpms installs the slpad.service file and its service releated files ..if we are planning to start this slapd.service then these packages needs to be installed.